CMakeLists.txt

```diff
@@ -122,22 +155,70 @@ if(CCPP)
     set(ENV{CCPP_CAPS} "${CAPS}")
     set(ENV{CCPP_TYPEDEFS} "${TYPEDEFS}")
 
-endif(CCPP)
+endif()
+
+###############################################################################
+### FMS
+###############################################################################
+include(fms_files.cmake)
+add_library(fms ${fms_src_files})
+# stupid cmake can not figure out dependency of fft.F90 on fft99.F90 because 'use fft99_mod' is inside ifdefs
+set_property(SOURCE FMS/fft/fft.F90 APPEND_STRING PROPERTY COMPILE_FLAGS "-DSGICRAY=0 -DNAGFFT=0")
+
+target_include_directories(fms PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/FMS/include>
+                                      $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/FMS/fms>
+                                      $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/FMS/fms2_io/include>
+                                      $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/FMS/mpp/include>)
+
+target_link_libraries(fms PRIVATE NetCDF::NetCDF_Fortran)
+set_target_properties(fms PROPERTIES ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/FMS)
+set_target_properties(fms PROPERTIES Fortran_MODULE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/FMS/mod)
+target_include_directories(fms PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/FMS/mod>)
+
+###############################################################################
+### CMEPS
+###############################################################################
+if(CMEPS)
```

Comment:
why do you need an if-block in S2S model.  Isn't that mandatory here?

Reply:
Will change to if(S2S) that will enable CICE6/MOM6/CMEPS, then when -weather adopts CMEPS we can make it permanent later